pkg/cmd/create/create_vault.go

```diff
@@ -326,34 +331,68 @@ func (o *CreateVaultOptions) createVaultGKE(vaultOperatorClient versioned.Interf
 
 func (o *CreateVaultOptions) createVaultAWS(vaultOperatorClient versioned.Interface, vaultName string,
 	kubeClient kubernetes.Interface, clusterName string, vaultAuthServiceAccount string) error {
-	if o.S3Bucket == "" {
-		return fmt.Errorf("Missing S3 bucket flag")
-	}
-	if o.KMSKeyID == "" {
-		return fmt.Errorf("Missing AWS KMS key id flag")
-	}
-	if o.AccessKeyID == "" {
-		return fmt.Errorf("Missing AWS access key id flag")
-	}
-	if o.SecretAccessKey == "" {
-		return fmt.Errorf("Missing AWS secret access key flag")
-	}
-	if o.DynamoDBRegion == "" || o.KMSRegion == "" || o.S3Region == "" {
+
+	if o.AutoCreate == true {
 		defaultRegion, err := amazon.ResolveRegionWithoutOptions()
 		if err != nil {
 			return errors.Wrap(err, "finding default AWS region")
 		}
-		log.Logger().Infof("Region not specified, defaulting to %s", util.ColorInfo(defaultRegion))
-		if o.DynamoDBRegion == "" {
-			o.DynamoDBRegion = defaultRegion
+
+		o.DynamoDBRegion = defaultRegion
+		o.KMSRegion = defaultRegion
+		o.S3Region = defaultRegion
+
+		domain := "jenkins-x-domain"
+		username := "vault_" + defaultRegion
+
+		accessKey, keySecret, kmsID, s3Name, tableName, err := awsvault.CreateVaultResources(awsvault.ResourceCreationOpts{
+			Region:    defaultRegion,
+			Domain:    domain,
+			Username:  username,
+			TableName: autoCreateTableName,
+		})
+
+		if err != nil {
+			return errors.Wrap(err, "An error occurred while creating the vault resources")
+		}
+
+		o.S3Bucket = *s3Name
```

Comment:
Is there any particular reason to use pointers here instead of plain strings?

Reply:
I initially tried using the return type `string` however that prevented me from returning `nil` as I got `cannot convert nil (untyped nil value) to string`. This is my first time at go so if there is a better/more standardish way at approaching this I can go ahead refactor.

Reply:
Usually, You can return an empty string `""` with an error instead of a pointer nil with an error. The error already signals that the returned strings are not valid.